toolkit/tools/pkg/imagecustomizerlib/partitionutils.go

```diff
@@ -352,24 +405,156 @@ func findPartition(mountIdType imagecustomizerapi.MountIdentifierType, mountId s
 	return partition, partitionIndex, nil
 }
 
+func convertDevToUuid(partitions []diskutils.PartitionInfo, buildDir string) (string, error) {
+	for _, partition := range partitions {
+		matches, err := checkExtendedDevPartition(partition, partitions, buildDir)
+		if err != nil {
+			return "", err
+		}
+		if matches {
+			return partition.Uuid, nil
+		}
+	}
+	return "", fmt.Errorf("unable to find UUID for /dev path")
+}
+
+// checkExtendedDevPartition checks whether a given partition is associated with a verity-enabled root partition.
+// This function is specific to the case where the partition type is "dev" (device path, e.g., /dev/mapper/root).
+// It is an extension of the findExtendedPartition logic and specializes in handling verity partitions,
+// supporting both UKI and non-UKI configurations by validating against the kernel cmdline arguments or grub configuration.
+// Returns true if the partition matches the expected verity configuration, otherwise false.
+func checkExtendedDevPartition(partition diskutils.PartitionInfo, partitions []diskutils.PartitionInfo,
```

Comment:
This function should be split up. You should have a function that extracts the kernel command-line, correctly picking between pulling from either UKI and grub.cfg (and eventually systemd-boot). Then another function that extracts the verity info from the kernel command-line

Reply:
Revised, could you please check it again? Thanks.